### ISSUE TITLE
Add support to jupyter module for BackendV2 and other providers

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit#9106]
### What should we add?

The jupyter module in `qiskit.tools.jupyter` has a bunch of magics and default formatters set for backends from the qiskit-ibmq-provider. However, that package has been superseded by qiskit-ibm-provider and will soon be deprecated and archived. This will make the jupyter functionality with backends not usable anymore. We should update all this functionality to be generic and work with any backend, not just those from the ibmq provider.

**Comment by ohadlev77** [comment]
Hey, I am interested working on this one.

For clarification - what we want is that the table generated for `BackendV1` (generated by code in `qiskit.tools.jupyter.backend_monitor`) upon calling a `BackendV1` object will work the same for `BackendV2` objects?

**Comment by mtreinish** [comment]
Yeah that's basically it, along with loosening the type hints to say they will accept any BackendV1 or BackendV2 based backend instead of being limited to `IBMQBackend` and `FakeBackend`. The other piece is updating the `HTML_FORMATER` hooks to work on the abstract backend types instead of importing the ibmq provider and setting it for that.

**Comment by 1ucian0** [comment]
Hi @ohadlev77 this still affects 0.46, which has extended support until August 15 (after that date, it can be closed). Are you still interested?
